packages/types/index.d.ts

```diff
@@ -22,3 +23,8 @@ export type PluginRpcHandler = (
 export interface PluginProvider extends MetaMaskInpageProvider {
   registerRpcMessageHandler: (handler: PluginRpcHandler) => void;
 }
+type PluginName = string;
+export interface ErrorMessageEvent {
+  type: 'error';
+  payload: [PluginName, { message: string; code: number; data?: Json }];
+}
```

Comment:
Yeah, we definitely need a different name (or rather, `type`) for this. It should convey what kind of error it is... Controllers use their `name` fields to namespace their events, but since this event is intended for any execution environment service, we need to come up with something else.

How about this?

```suggestion
export interface ErrorMessageEvent {
  type: 'PluginExecutionService:unhandledError';
  payload: [PluginName, { message: string; code: number; data?: Json }];
}
```

We should probably create an `enum` for storing all events like this.